gitlab/client_repository_commit.go

```diff
@@ -70,16 +70,17 @@ func (c *CommitClient) Create(_ context.Context, branch string, message string,
 		return nil, fmt.Errorf("no files added")
 	}
 
-	fileAction := gitlab.FileCreate
-
 	commitActions := make([]*gitlab.CommitActionOptions, 0)
 	for _, file := range files {
-		filePath := *file.Path
-		fileContent := *file.Content
+		fileAction := gitlab.FileCreate
+		if file.Content == nil {
+			fileAction = gitlab.FileDelete
```

Comment:
Can we instead Have a new method that does the deletion?

Reply:
Ah! What are you thinking?

Ideally we want to support creating and deleting and chmoding etc in a single commit...

We could add a `mode` attr to `type CommitFile struct`, so you could declare deletions / other operations when creating a commit..

Reply:
- Github uses: [Content == nil ](https://github.com/google/go-github/blob/master/github/git_trees.go#L134)
- Gitlab uses: "action" `gitlab.FileCreate | gitlab.FileDelete`

So I guess we should decide which way we want to take the ggp api..

Reply:
If we went for `nil` checking for now... and then added support for `Action` later, for `chmod`ing and that.. We'd have 2 x ways of doing something which is a bit awkward. If you supplied "Content == nil && gitproviders.FileCreate` (if we added gitproviders.FileCreate in the future), then we'd have to have some nice error message explaining it doesn't make sense.

However we have github code out in the wild now that is deleting files w/ `Content == nil`, so that would Just Work with gitlab now which is nice. This seems like an okayish way foward for now, wdyt?